### PR TITLE
Update SmallRye GraphQL to 1.0.12

### LINKED ISSUE #12302: GraphQL in quarkus
**Describe the bug**
Seems that GraphQL in quarkus breaks if we wander a bit outside the path described in the guide. If I add a main graphql resource in order to build a larger application with a lot of different objects/paths things seems to fall apart. 

Suddenly all that works is "allFilms". "getFilm" doesn't exist and I haven't looked too much into the other tests in the getting started project. 

https://github.com/mejlholm/quarkus-quickstarts/tree/master/microprofile-graphql-quickstart

**Expected behavior**
Tests passing

**Actual behavior**
Tests fail

**To Reproduce**

https://github.com/mejlholm/quarkus-quickstarts/tree/master/microprofile-graphql-quickstart

**Configuration**
Nothing 

**Screenshots**

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
Microsoft Windows [Version 10.0.19041.508]

 - Output of `java -version`: 
java version "1.8.0_261"
Java(TM) SE Runtime Environment (build 1.8.0_261-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.261-b12, mixed mode)

 - Quarkus version or git rev: 
1.8.1.Final

 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\dev\tools\apache-maven-3.6.3\bin\..
Java version: 14.0.1, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-14.0.1
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

**Additional context**
(Add any other context about the problem here.)

### LINKED ISSUE #12402: SmallRye GraphQL conflict with same enum names in different models
**SmallRye GraphQL cannot differentiate same enum names in different models from version 1.6.0 to 1.8.1.**
We have GraphQL API for banking. We have separate models i.e `Accounts` and `Balances`. In `Accounts`  we have an enum called `Type` and we have enum with same name `Type` in `Balances`.

**The expected behavior is to see they're working fine as in previous versions.**
The quarkus version was 1.5.2 for this apps and we decide to update the quarkus and plug-in versions. In the all versions after 1.5.2, the problem is same. In 1.5.2, it perfectly works without having conflict even if we have same enum name in different models.

**Actual behavior**
 The data is stored in Neo4j and when we run the application, it works fine with accounts. But when running the `Balances`, it gives the following stacktrace:
```
2020-09-29 18:51:46,311 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:134)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:60)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
        ... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:287)
        ... 15 more
Caused by: graphql.AssertException: type can't be null
        at graphql.Assert.assertNotNull(Assert.java:16)
        at graphql.schema.GraphQLFieldDefinition.<init>(GraphQLFieldDefinition.java:85)
        at graphql.schema.GraphQLFieldDefinition$Builder.build(GraphQLFieldDefinition.java:473)
        at io.smallrye.graphql.bootstrap.Bootstrap.createGraphQLFieldDefinitionFromField(Bootstrap.java:399)
        at io.smallrye.graphql.bootstrap.Bootstrap.createGraphQLFieldDefinitionsFromFields(Bootstrap.java:386)
        at io.smallrye.graphql.bootstrap.Bootstrap.createGraphQLObjectType(Bootstrap.java:284)
        at io.smallrye.graphql.bootstrap.Bootstrap.createGraphQLObjectTypes(Bootstrap.java:271)
        at io.smallrye.graphql.bootstrap.Bootstrap.generateGraphQLSchema(Bootstrap.java:123)
        at io.smallrye.graphql.bootstrap.Bootstrap.bootstrap(Bootstrap.java:103)
        at io.smallrye.graphql.cdi.producer.GraphQLProducer.initialize(GraphQLProducer.java:41)
        at io.smallrye.graphql.cdi.producer.GraphQLProducer.initialize(GraphQLProducer.java:37)
        at io.smallrye.graphql.cdi.producer.GraphQLProducer_ClientProxy.initialize(GraphQLProducer_ClientProxy.zig:185)
        at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLRecorder.createExecutionService(SmallRyeGraphQLRecorder.java:24)
        at io.quarkus.deployment.steps.SmallRyeGraphQLProcessor$buildExecutionService-1556698146.deploy_2(SmallRyeGraphQLProcessor$buildExecutionService-1556698146.zig:5278)
        at io.quarkus.deployment.steps.SmallRyeGraphQLProcessor$buildExecutionService-1556698146.deploy(SmallRyeGraphQLProcessor$buildExecutionService-1556698146.zig:50)
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:267)
```

**To Reproduce**
1. Create 2 separate models.
2. Have an enum type in both models with same enum name.(i.e enum `Type` in `Accounts` and `Balances`)
3. Run it in debug mode, put the breakpoint on line 85 in `GraphQLFieldDefinition.java` and see the hitting `type can't be null` on same enum name.

**Environment (please complete the following information):**
 - MSYS_NT-10.0-19041 DESKTOP-6RLSJFU 3.0.7-338.x86_64 2019-11-21 23:07 UTC x86_64 Msys
 - Output of `java -version`: openjdk version "11.0.7" 2020-04-14
 - GraalVM version (if different from Java): OpenJDK Runtime Environment GraalVM CE 20.1.0 (build 11.0.7+10-jvmci-20.1-b02)
 - Quarkus version or git rev: 1.8.1
 - Apache Maven 3.6.3

**Additional context**
If you really want to see that's the reason, you can simply create other models as well and have same enum name. It will hit on that and will say `type can't be null`

Regards!
Yusuf

### LINKED ISSUE #12452: Packaging native image fails with SmallRye GraphQL
**Describe the bug**
When using SmallRye GraphQL, packaging a native image fails with the following error:

> com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Class initialization of io.smallrye.graphql.cdi.validation.ValidationService failed ... Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.smallrye.graphql.cdi.validation.ValidationService

This error is solved with the following workaround in application.properties:
> quarkus.native.additional-build-args= --initialize-at-run-time=io.smallrye.graphql.cdi.validation.ValidationService

Buty I guess this not a normal situation.

**Expected behavior**
>  mvn package -Pnative
working

**Actual behavior**
>  mvn package -Pnative
failing

**To Reproduce**
This [GitHub project](https://github.com/jefrajames/demobugValidationService.git) reproduces the error.

**Configuration**
```properties
# Add your application.properties here, if applicable.
quarkus.native.additional-build-args= --initialize-at-run-time=io.smallrye.graphql.cdi.validation.ValidationService
```

**Environment (please complete the following information):**

* macOS 10.13.6
* OpenJDK 11.0.7
* GraalVM CE 20.20.0
* Maven 3.6.3

**Additional context**
It works in Java mode.

### PR DESCRIPTION
**Pull request by phillip-kruger**
This PR upgrades SmallRye GraphQL to 1.0.12.

The following is included:

- Fix #12452 - Native image problem with ValidationService (@jefrajames)
- Fix #12402 - Added naming strategies for types (@miador)
- Fix #12302 - Allow grouping of top-level endpoint with `@Name` (@mejlholm)
- Better support for generics (@velias)

Also see https://github.com/smallrye/smallrye-graphql/releases/tag/1.0.12

Signed-off-by:Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

File: extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLConfig.java [hunk 1]
```diff
@@ -19,6 +20,12 @@
     @ConfigItem(name = "metrics.enabled", defaultValue = "false")
     boolean metricsEnabled;
 
+    /**
+     * Change the type naming strategy. Default, MergeInnerClass, Full
```
Comment (gsmet) [inline extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLConfig.java:24]: Is `TypeAutoNameStrategy` an enum? If so you can remove the default values, they will be automatically included in the doc.

File: extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java [hunk 2]
```diff
@@ -293,6 +297,15 @@ void openTracingIntegration(Capabilities capabilities,
         return classes;
     }
 
+    private Set<String> getOperationClassNames(Map<Group, Set<Operation>> groupesOperations) {
```
Comment (gsmet) [inline extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java:300]:
```suggestion
    private Set<String> getOperationClassNames(Map<Group, Set<Operation>> groupedOperations) {
```
?

File: extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/AbstractGraphQLTest.java [hunk 3]
```diff
@@ -61,8 +62,14 @@ protected JsonObject createRequestBody(String graphQL, JsonObject variables) {
     }
 
     protected static String getPropertyAsString() {
+        return getPropertyAsString(null);
+    }
+
+    protected static String getPropertyAsString(Map<String, String> otherProperties) {
         try {
             StringWriter writer = new StringWriter();
+            if (otherProperties != null)
```
Comment (gsmet) [inline extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/AbstractGraphQLTest.java:71]: can you add curly braces?

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLConfig.java (hunk 1 above)
Comment (phillip-kruger) [reply extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLConfig.java]: Cool. Thanks :)

**Review by gsmet — APPROVED** [review]

**Review by miador — APPROVED** [review]

**Review by miador — APPROVED** [review]

**Comment by gsmet** [comment]
Rebased and force pushed. Let's wait for CI.

**Comment by phillip-kruger** [comment]
Thanks @gsmet !

**Pull request merged**